File: BOJ(Java 공부용)/2309.java
import java.io.*;
import java.util.*;
public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] arr = new int[9];
        boolean[] visited = new boolean[9];
        for(int i=0;i<9;i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            arr[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(arr);
        go(0,0,0,arr,visited);

    }
    private static void go(int here,int cnt,int sum,int[] arr,boolean[] visited) throws IOException {
        if(here>=10) return;
        if(cnt == 7) {
            if(sum == 100) {
                for(int i=0;i<9;i++) {
                    if(visited[i]) bw.write(arr[i]+"\n");
                }
                bw.flush();
                bw.close();
                System.exit(0);
            }
            return;
        }
        for(int i=here;i<9;i++) {
            if(!visited[i]) {
                visited[i] = true;
                go(i+1,cnt+1,sum+arr[i],arr,visited);
                visited[i] = false;
            }
        }
    }
}

File: BOJ(Python 공부용)/2309.py
import sys

input = sys.stdin
sys.setrecursionlimit(5000)
v = []
temp = []
visited = [False] * 9


def go(here, cnt, depth):
    if cnt == depth:
        sum = 0
        for num in temp:
            sum += num

        if sum != 100: return
        for num in temp:
            print(num)
        sys.exit(0)
    for i in range(here, 9, 1):
        if not visited[i]:
            visited[i] = True
            temp.append(v[i])
            go(i + 1, cnt + 1, depth)
            temp.remove(v[i])
            visited[i] = False


for _ in range(9):
    v.append(int(input.readline()))

v.sort()
go(0, 0, 7)